lib/ansible/template/__init__.py

```diff
@@ -518,8 +519,20 @@ def __getitem__(self, key):
                 acr = AnsibleCollectionRef.from_fqcr(ref=redirect_fqcr, ref_type=self._dirname)
                 display.vvv('redirecting {0} {1} to {2}.{3}'.format(self._dirname, key, acr.collection, acr.resource))
                 key = redirect_fqcr
-            # TODO: handle recursive forwarding (not necessary for builtin, but definitely for further collection redirs)
-
+                acr1 = AnsibleCollectionRef.try_parse_fqcr(key, self._dirname)
+                if not acr1:
+                    pass
+                else:
+                    ts = _get_collection_metadata(acr.collection)
+                    leaf_key = key.split('.')[-1]
+                    routing_entry = ts.get('plugin_routing', {}).get(self._dirname, {}).get(leaf_key, {})
+                    redirect_fqcr1 = routing_entry.get('redirect', None)
+                    if redirect_fqcr1:
```

Comment:
```suggestion
                    if redirect_fqcr1 := routing_entry.get('redirect', None)
```

Also, it would be good to use a better name than redirect_fqcr1.